src/main/java/seedu/address/commons/util/LoanSlipUtil.java

```diff
@@ -0,0 +1,217 @@
+        File file = new File(finalDest);
+        file.getParentFile().mkdirs();
+        PdfWriter writer = new PdfWriter(finalDest);
+        //Initialize PDF document
+        PdfDocument pdf = new PdfDocument(writer);
+        // Initialize document
+        currentFile = file;
+        return new Document(pdf);
+    }
+
+    /**
+     * Populates the pdf file with information, creating the loan slip.
+     *
+     * @param doc {@code LoanSlipDocument} object to be populated with data.
+     */
+    private static void generateLiberryLoanSlip(LoanSlipDocument doc) throws LoanSlipException {
+        if (!isMounted) {
+            throw new LoanSlipException("No Loan slip mounted yet");
+        }
+        //write logo
+        doc.writeLogo();
+        doc.writeLine();
+        //Add paragraph to the document
+        doc.writeHeader(currentBorrower.getName().toString());
+        doc.writeLeftParagraph(currentBorrower.getBorrowerId().toString());
+        doc.writeLine();
+        doc.writeMidHeader("Books borrowed");
+        //populate table
+        String[] headerRow = new String[]{"S/N", "Book", "Due By"};
+        doc.writeRow(headerRow);
+        doc.writeRow(createBookRow());
+        //add table to document
+        doc.submitTable();
+        doc.writeLine();
+        doc.writeCentralisedParagraph(BYE_MESSAGE);
+        //close document
+        doc.closeDoc();
+        isGenerated = true;
+    }
+
+    /**
+     * Creates an array using the details from the mounted loan slip.
+     *
+     * @return an array of string representing a row of the table.
+     */
+    private static String[] createBookRow() throws LoanSlipException {
+        if (!isMounted) {
+            throw new LoanSlipException("No Loan slip mounted yet");
+        }
+        String[] currentBookDetails = new String[3];
+        currentBookDetails[0] = currentBook.getSerialNumber().toString();
+        currentBookDetails[1] = currentBook.getTitle().toString();
+        currentBookDetails[2] = formatDate(currentLoan.getDueDate().toString());
+        return currentBookDetails;
+    }
+
+    /**
+     * Formats the date to be printed on the loan slip.
+     *
+     * @param date date to be formatted.
+     * @return a formatted date as a {@code String}
+     */
+    private static String formatDate(String date) {
+        String[] pieces = date.split("-");
+        String result = "";
+        for (String s : pieces) {
+            result = s + "-" + result;
+        }
+        return result.substring(0, result.length() - 1);
+    }
+
+    /**
+     * Returns true if a loan slip is currently mounted.
+     *
+     * @return true if a loan slip is currently mounted.
+     */
+    public static boolean isMounted() {
+        return isMounted;
+    }
+
+    /**
+     * Opens loan slip pdf to allow ease of printing loan slip.
+     * (Cannot be tested autonomously)
+     */
+    public static void openLoanSlip() throws LoanSlipException {
```

Comment:
Maybe we can have another method next time to print the loan slip by using the method java.awt.Desktop.print(File)

Reply:
Will consider this alternative next time when we really want to print it out and test 😂

Reply:
But current implementation allows for this flexibility to insert this feature with ease :)